### PR TITLE
fix: Fixed Qarnot image upload and image retrieval

### PR DESCRIPTION
**Pull request by florianriche**
Upload routes were not working properly with Qarnot:

1. For the upload: Qarnot doesn't accept an int as a bucket key. Had to convert the hash to a string
2. For the image URL retrieval: Qarnot retrieves the bucket objects by downloading them locally and then simply returning the filepath of the object. It seems here is not way to retrieve a public URL. Hence I didn't touch this part, but we have to be aware it is malfunctioning with Qarnot as it will only return the filename (i.e the bucket_key, which is not very useful)
3. For the image retrieval as Bytes: Used to try to read a URL. Now it reads the Bucket File downloaded by Qarnot, uses it in the Response and launch a Background task to delete that file after the route response..

### CODE REVIEW

File: src/app/services/bucket/qarnotBucketService.py [hunk 1]
```diff
@@ -13,10 +14,23 @@ def connect_to_bucket(self):
...
+            self.connect_to_bucket().add_file(file_binary, bucket_key)
+        except Exception as e:
+            print(e)
+            return False
+        return True
 
     async def get_uploaded_file(self, bucket_key: str):
-        return await self.connect_to_bucket().get_file(bucket_key)
+        try:
+            return self.connect_to_bucket().get_file(bucket_key)
```
Comment (martin1tab) [inline src/app/services/bucket/qarnotBucketService.py:26]: No need for await ?

**Review by martin1tab — APPROVED** [review]

File: src/app/services/bucket/qarnotBucketService.py (hunk 1 above)
Comment (florianriche) [reply src/app/services/bucket/qarnotBucketService.py:26]: I got an error stating if I set it. as get_file returns a string a got a `object str can't be used in 'await' expression`

File: src/app/api/routes/media.py [hunk 2]
```diff
@@ -93,7 +90,7 @@ async def upload_media(media_id: int = Path(..., gt=0),
     Upload a media (image or video) linked to an existing media object in the DB
     """
     entry = await check_for_media_existence(media_id, current_device.id)
-    bucket_key = hash(datetime.utcnow())
+    bucket_key = str(hash(datetime.utcnow()))
```
Comment (frgfm) [inline src/app/api/routes/media.py:93]: Good catch :+1:

File: src/app/services/bucket/qarnotBucketService.py [hunk 3]
```diff
@@ -13,10 +14,23 @@ def connect_to_bucket(self):
         return self.bucket
 
     async def upload_file(self, bucket_key: str, file_binary: bin):
-        return await self.connect_to_bucket().add_file(file_binary, bucket_key)
+        try:
+            self.connect_to_bucket().add_file(file_binary, bucket_key)
+        except Exception as e:
+            print(e)
```
Comment (frgfm) [inline src/app/services/bucket/qarnotBucketService.py:20]: Perhaps a warning or logging would be better for the exception?

**Review by frgfm — APPROVED** [review]
Thanks for the quick fix! Just a few questions, but nothing important, happy to merge if you already have the answers

File: src/app/services/bucket/qarnotBucketService.py (hunk 3 above)
Comment (florianriche) [reply src/app/services/bucket/qarnotBucketService.py:20]: Like 
```python
import logging
logging.basicFongi()
logging.error(e) ? 
```

Comment (frgfm) [reply src/app/services/bucket/qarnotBucketService.py:20]: Yup! But regarding the logger choice, I'd recommend the uvicorn logger
```
logger = logging.getLogger("uvicorn.error")
```

**Review by frgfm — APPROVED** [review]
Thanks a lot for this save! Can you confirm whether it solved the problem on your end?

**Comment by florianriche** [comment]
yes, on my sied I managed to upload and retriee Qarnot images using  swagger.

**Comment by florianriche** [comment]
The clients tests and e2e are failing though.

**Comment by frgfm** [comment]
> The clients tests and e2e are failing though.

The e2e test seems to be working, but the failing of the client test is not related! We can move forward with merging I think :+1:

**Pull request merged**